openthread/cmake/extensions.cmake

```diff
@@ -26,7 +50,61 @@ endif()
   endforeach()
 
   list(SORT OPENTHREAD_SETTINGS)
-  list(INSERT OPENTHREAD_SETTINGS 0 "OpenThread_commit=${openthread_git_hash}\n")
+
+  find_package(Git QUIET)
+  if(GIT_FOUND)
+    get_mbedtls_dir(ARM_MBEDTLS_PATH)
+    get_git_hash(${ARM_MBEDTLS_PATH})
+    list(INSERT OPENTHREAD_SETTINGS 0 "MBEDTLS_commit=${git_hash}\n")
+
+    get_git_hash(${ZEPHYR_NRFXLIB_MODULE_DIR})
+    list(INSERT OPENTHREAD_SETTINGS 0 "NRFXLIB_commit=${git_hash}")
+
+    get_git_hash(${ZEPHYR_OPENTHREAD_MODULE_DIR})
+    list(INSERT OPENTHREAD_SETTINGS 0 "OpenThread_commit=${git_hash}")
+  endif()
+  
   FILE(WRITE ${CONFIG_FILE} ${OPENTHREAD_SETTINGS})
+endfunction()
+
+function(get_active_mbedtls_configs_from_file fileName returnMatch1List)
+  file(READ ${fileName} FILE_CONTENT)
+  set(PATTERN "[ \t\n]#define[ \t]+(MBEDTLS_[A-Za-z0-9_]*)")
+  string(REGEX MATCHALL ${PATTERN} match_list ${FILE_CONTENT})
+    foreach(element ${match_list})
+      string(REGEX MATCH ${PATTERN} MACRO_DEF ${element})
+      list(APPEND match1_list "${CMAKE_MATCH_1}\n")
+      set(${returnMatch1List} ${match1_list} PARENT_SCOPE)
+    endforeach()
+endfunction()
+
+function(check_openthread_dependencies ot_lib_nrf_security_mbedtls_config_file)
+  set(nrf_security_mbedtls_config_file "${CMAKE_CURRENT_BINARY_DIR}/../nrf_security/include/${CONFIG_MBEDTLS_CFG_FILE}")
```

Comment:
This is checking the `CMAKE_CURRENT_BINARY_DIR`, but what if the build directory is empty? This is failing for me:
```

cd nrf/samples/openthread/cli
rm -rf build
west build -b nrf52840dk_nrf52840 -t install_openthread_libraries
rm -rf build
west build -b nrf52840dk_nrf52840 -- -DOVERLAY_CONFIG="harness/overlay-cert.conf"
```

Reply:
It can't be empty. The referential file is created just before checking if only nrf_security is enabled.

Reply:
What you mean with "_if only nrf_security is enabled_"? `CONFIG_OPENTHREAD_NRF_SECURITY` is enabled by default for OT samples, but with the previous steps I got this error:
```
/home/edmo/vscode/ncs/nrfxlib/openthread/lib/cortex-m4/soft-float/v1.1/master//nrf_security_mbedtls_configuration.h
/home/edmo/vscode/ncs/nrf/samples/openthread/cli/build/modules/nrfxlib/openthread/../nrf_security/include/
CMake Error at /home/edmo/vscode/ncs/nrfxlib/openthread/cmake/extensions.cmake:71 (file):
  file failed to open for reading (No such file or directory):

    /home/edmo/vscode/ncs/nrf/samples/openthread/cli/build/modules/nrfxlib/openthread/../nrf_security/include/
Call Stack (most recent call first):
  /home/edmo/vscode/ncs/nrfxlib/openthread/cmake/extensions.cmake:85 (get_active_mbedtls_configs_from_file)
  /home/edmo/vscode/ncs/nrfxlib/openthread/CMakeLists.txt:54 (check_openthread_dependencies)


-- Configuring incomplete, errors occurred!
```

Reply:
@edmont Could you check if it works for you now? Build from here: https://github.com/nrfconnect/sdk-nrf/pull/2974